File: 백준/Silver/1978. 소수 찾기/소수 찾기.py
N = int(input())

result = 0

num = input().split()

for i in range(N) :
    num[i] = int(num[i])
    
    if num[i] == 1 :
        result += 1
    else :
        for j in range(2, int(num[i] ** (1/2)) + 1) :
            if num[i] % j == 0 :
                result += 1
                break

print(N - result)

File: 백준/Silver/1978. 소수 찾기/소수 찾기.java
import java.io.*;
import java.util.*;

public class Main {
	public static void main(String[] args) throws IOException {
		Scanner sc = new Scanner(System.in);
		
		int N = sc.nextInt();
		
		int[] num = new int[1001];
		
		int result = 0;
		num[0] = 1;
		num[1] = 1;
		
		for (int j = 2; j <= 1000; j++) {
			int point = 2;
			
			if (num[j] == 0) {
				while (j * point <= 1000) {
					num[j * point] = 1;
					point++;
				}
			}
		}
		
		for (int i = 0; i < N; i++) {
			int checknum = sc.nextInt();
			
			if (num[checknum] == 0) {
				result++;
			}
		}
		System.out.println(result);
		
		sc.close();
	}
}